com.unity.ml-agents.extensions/Runtime/Match3/AbstractBoard.cs

```diff
@@ -4,7 +4,10 @@
 
 namespace Unity.MLAgents.Extensions.Match3
 {
-    public abstract class AbstractBoard : MonoBehaviour
+    /// <summary>
+    /// Representation of the AbstractBoard size and number of cell and special types.
+    /// </summary>
+    public struct BoardSize
```

Comment:
I'd like review/feedback on this interface before I update the documentation/migration/changelog.